cmake/macros.cmake

```diff
@@ -1296,12 +1298,13 @@ FUNCTION( FINALIZE_TESTBUILDER )
     ENDIF()
     # If we are using HIP and COMPILE_CXX_AS_HIP is set, change the language
     IF ( USE_HIP AND COMPILE_CXX_AS_HIP )
-        FOREACH ( tmp ${TESTBUILDER_SOURCES} )
+        FOREACH ( tmp ${TB_TARGET_SOURCES} )
```

Comment:
@rbberger, Could lines 1278-279 be deleted and 1300 be modified to be 

 IF ( TESTBUILDER_SOURCES AND USE_HIP AND COMPILE_CXX_AS_HIP )

Reply:
No. The problem is when ` TESTBUILDER_SOURCES` is empty, nothing sets the necessary HIP language for `TestBuilder.cpp`.

Reply:
Oh wait, in that case also the linker language stays unmodified. Let me try.

Reply:
Nope, the flags are injected by Kokkos. So unless you remove the kokkos dependence in that case too, you will get that error.
```
g++: error: unrecognized command-line option '-fno-gpu-rdc'
g++: error: unrecognized command-line option '--offload-arch=gfx90a'
```

Reply:
Hmm -- I think the intent was that in the case TESTBUILDER_SOURCES is empty TestBuilder.cpp is not compiled. But I guess you are saying that it is?

Reply:
Yes, see https://github.com/AdvancedMultiPhysics/AMP/blob/master/src/operators/test/CMakeLists.txt#L150-L179. We're only building with `+mpi+kokkos+rocm`.

Reply:
The executable is always added during [INITIALIZE_TESTBUILDER](https://github.com/AdvancedMultiPhysics/TPL-builder/blob/master/cmake/macros.cmake#L1248-L1252). To get the behavior you outlined above, you would need to move to FINALIZE_TESTBUILDER.

Reply:
Okay, sounds good. Mark will change if he wants that behavior